geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/planning/guard/TemporalQueryGuardTest.scala

```diff
@@ -0,0 +1,79 @@
+/***********************************************************************
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.index.planning.guard
+
+import org.geotools.data.{Query, Transaction}
+import org.geotools.filter.text.ecql.ECQL
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.index.TestGeoMesaDataStore
+import org.locationtech.geomesa.utils.collection.SelfClosingIterator
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class TemporalQueryGuardTest extends Specification {
+
+
+  "TemporalQueryGuard" should {
+    "block queries with an excessive duration" in {
+      // note: z3 needs to be declared first so it's picked for full table scans
+      val sft = SimpleFeatureTypes.createType("c4f4ef29-6e41-4113-9b74-adf35711aa7a",
+        "name:String,age:Int,dtg:Date,*geom:Point:srid=4326;geomesa.indices.enabled='z3,id,attr:name'")
+      sft.getUserData.put("geomesa.query.interceptors",
+        "org.locationtech.geomesa.index.planning.guard.TemporalQueryGuard");
+      sft.getUserData.put("geomesa.guard.temporal.max.duration", "1 day")
+
+      val ds = new TestGeoMesaDataStore(true)
+      ds.createSchema(sft)
+
+      val valid = Seq(
+        "name = 'bob'",
+        "IN('123')",
+        "bbox(geom,-10,-10,10,10) AND dtg during 2020-01-01T00:00:00.000Z/2020-01-01T23:59:59.000Z",
+        "bbox(geom,-10,-10,10,10) AND (dtg during 2020-01-01T00:00:00.000Z/2020-01-01T00:59:59.000Z OR dtg during 2020-01-01T12:00:00.000Z/2020-01-01T12:59:59.000Z)"
+      )
+
+      val invalid = Seq(
+        "INCLUDE",
+        "bbox(geom,-10,-10,10,10)",
+        "bbox(geom,-180,-90,180,90)",
```

Comment:
Ok, what changed that these are getting blocked now?

Reply:
i swapped the order of the indices in the user data definition - the first index declared gets used for full table scans (which may be an unintentional implementation detail...). i left a comment on that change up above; that made the full table scans hits the z3 index and get blocked by the guard, instead of the feature id index where they weren't.

not sure if we should do something to prioritize which index gets used for full table scans - when we switched to SPI loaded indices we lost the ability to easily configure that, but we could add something like a 'priority' flag to each index. or just document that you should declare them in order... in practice i think almost all serious installations would block full tables scans though?

Reply:
Wow.  That's an interesting set of choices.

Do we have an analog to this page: https://docs.geoserver.org/stable/en/user/production/index.html ?

Any thoughts about which of these would be easiest for the end users?

Reply:
i don't think we have anything like that - seems like a good idea.

i'm not sure which would be better... on the one hand, being able to specify a table to use for full table scans seems like it might be nice. on the other, having inconsistent tables being scanned seems potentially confusing if you don't know what's going on.